packages/vuetify/src/components/VOtpInput/VOtpInput.ts

```diff
@@ -0,0 +1,343 @@
+          maxlength: 1,
+        },
+        on: Object.assign(listeners, {
+          blur: this.onBlur,
+          input: (e: Event) => this.onInput(e, otpIdx),
+          focus: (e: Event) => this.onFocus(e, otpIdx),
+          paste: (e: ClipboardEvent) => this.onPaste(e, otpIdx),
+          keydown: this.onKeyDown,
+          keyup: (e: KeyboardEvent) => this.onKeyUp(e, otpIdx),
+        }),
+        ref: `input-${otpIdx}`,
+      })
+    },
+    genTextFieldSlot (otpIdx: number): VNode {
+      return this.$createElement('div', {
+        staticClass: 'v-text-field__slot',
+      }, [
+        this.genInput(otpIdx),
+      ])
+    },
+    onBlur (e?: Event) {
+      this.isFocused = false
+      e && this.$nextTick(() => this.$emit('blur', e))
+    },
+    onClick (otpIdx: number) {
+      if (this.isFocused || this.isDisabled || !this.$refs[`input-${otpIdx}`]) return
+
+      this.onFocus(undefined, otpIdx)
+    },
+    onFocus (e?: Event, otpIdx?: number) {
+      e?.preventDefault()
+      e?.stopPropagation()
+      const refKey = 'input-' + otpIdx
+      if (!this.$refs[refKey]) return
+
+      const input = this.$refs[refKey] as HTMLInputElement
+      if (document.activeElement !== this.$refs[refKey]) {
+        input.focus()
+        return input.select()
+      }
+
+      if (!this.isFocused) {
+        this.isFocused = true
+        input.select()
+        e && this.$emit('focus', e)
+      }
+    },
+    onInput (e: Event, otpIdx: number) {
+      const target = e.target as HTMLInputElement
+      const value = target.value
+      this.applyValue(otpIdx, target.value, () => {
+        this.internalValue = this.otp.join('')
+      })
+      this.badInput = target.validity && target.validity.badInput
+
+      const nextIndex = otpIdx + 1
+      if (value) {
+        if (nextIndex < this.length) {
+          this.changeFocus(nextIndex)
+        } else {
+          this.clearFocus(otpIdx)
+          this.onCompleted()
+        }
+      }
+    },
+    clearFocus (index: number) {
+      const input = this.$refs[`input-${index}`] as HTMLInputElement
+      input.blur()
+    },
+    onKeyDown (e: KeyboardEvent) {
+      if (e.keyCode === keyCodes.enter) {
+        this.$emit('change', this.internalValue)
+      }
+
+      this.$emit('keydown', e)
+    },
+    onMouseDown (e: Event, otpIdx: number) {
+      // Prevent input from being blurred
+      if (e.target !== this.$refs[`input-${otpIdx}`]) {
+        e.preventDefault()
+        e.stopPropagation()
+      }
+
+      VInput.options.methods.onMouseDown.call(this, e)
+    },
+    onMouseUp (e: Event, otpIdx: number) {
+      if (this.hasMouseDown) this.focus(e, otpIdx)
+
+      VInput.options.methods.onMouseUp.call(this, e)
+    },
+    onPaste (event: ClipboardEvent, index: number) {
+      const maxCursor = this.length - 1
+      // const clipboardData = event?.clipboardData?.getData('Text') ||
+      //   window?.clipboardData ||
+      //   event?.originalEvent.clipboardData
```

Comment:
Does this need to be here?

Reply:
Cleanup applied on upcoming commit.
- The comment includes fallbacks for clipboard data - lint does not allow unknown attributes on window and event.